src/components/drawing/symbol_defs.js

```diff
@@ -10,544 +11,798 @@ var d3 = require('@plotly/d3');
+            return align(angle, standoff, 'M0,0L-' + rx + ',-' + ry + 'V' + ry + 'Z');
         },
         noDot: true
     },
     'arrow-bar-up': {
         n: 49,
-        f: function(r) {
-            var rx = d3.round(r, 2);
-            var ry = d3.round(r * 2, 2);
-            return 'M-' + rx + ',0H' + rx + 'M0,0L-' + rx + ',' + ry + 'H' + rx + 'Z';
+        f: function(r, angle, standoff) {
+            if(skipAngle(angle)) return emptyPath;
+
+            var rx = round(r, 2);
+            var ry = round(r * 2, 2);
+            return align(angle, standoff, 'M-' + rx + ',0H' + rx + 'M0,0L-' + rx + ',' + ry + 'H' + rx + 'Z');
         },
         needLine: true,
         noDot: true
     },
     'arrow-bar-down': {
         n: 50,
-        f: function(r) {
-            var rx = d3.round(r, 2);
-            var ry = d3.round(r * 2, 2);
-            return 'M-' + rx + ',0H' + rx + 'M0,0L-' + rx + ',-' + ry + 'H' + rx + 'Z';
+        f: function(r, angle, standoff) {
+            if(skipAngle(angle)) return emptyPath;
+
+            var rx = round(r, 2);
+            var ry = round(r * 2, 2);
+            return align(angle, standoff, 'M-' + rx + ',0H' + rx + 'M0,0L-' + rx + ',-' + ry + 'H' + rx + 'Z');
         },
         needLine: true,
         noDot: true
     },
     'arrow-bar-left': {
         n: 51,
-        f: function(r) {
-            var rx = d3.round(r * 2, 2);
-            var ry = d3.round(r, 2);
-            return 'M0,-' + ry + 'V' + ry + 'M0,0L' + rx + ',-' + ry + 'V' + ry + 'Z';
+        f: function(r, angle, standoff) {
+            if(skipAngle(angle)) return emptyPath;
+
+            var rx = round(r * 2, 2);
+            var ry = round(r, 2);
+            return align(angle, standoff, 'M0,-' + ry + 'V' + ry + 'M0,0L' + rx + ',-' + ry + 'V' + ry + 'Z');
         },
         needLine: true,
         noDot: true
     },
     'arrow-bar-right': {
         n: 52,
-        f: function(r) {
-            var rx = d3.round(r * 2, 2);
-            var ry = d3.round(r, 2);
-            return 'M0,-' + ry + 'V' + ry + 'M0,0L-' + rx + ',-' + ry + 'V' + ry + 'Z';
+        f: function(r, angle, standoff) {
+            if(skipAngle(angle)) return emptyPath;
+
+            var rx = round(r * 2, 2);
+            var ry = round(r, 2);
+            return align(angle, standoff, 'M0,-' + ry + 'V' + ry + 'M0,0L-' + rx + ',-' + ry + 'V' + ry + 'Z');
         },
         needLine: true,
         noDot: true
+    },
+    'arrow': {
+        n: 53,
+        f: function(r, angle, standoff) {
+            if(skipAngle(angle)) return emptyPath;
+
+            var headAngle = PI / 2.5; // 36 degrees - golden ratio
+            var x = 2 * r * cos(headAngle);
+            var y = 2 * r * sin(headAngle);
+
+            return align(angle, standoff,
+                'M0,0' +
+                'L' + -x + ',' + y +
+                'L' + x + ',' + y +
+                'Z'
+            );
+        },
+        backoff: 0.95,
```

Comment:
Can we have a slightly smaller backoff here, or somehow round down the final pixel backoff to get a little overlap? If the line and arrow are large and the same color there's a slight visual gap from antialiasing
<img width="106" alt="Screen Shot 2022-09-26 at 08 23 34" src="https://user-images.githubusercontent.com/2678795/192275728-3d9ff273-82d5-4f78-a539-55909aadb7d3.png">

Reply:
Addressed in https://github.com/plotly/plotly.js/pull/6297/commits/dfd34d686e3710ba1f6bc438f2fd8459343c87d5.
Do you want me to tweak/add a test with large markers and thick lines?